csharp/src/Ice/IncomingFrame.cs

```diff
@@ -33,10 +33,8 @@ public abstract class IncomingFrame
                         var binaryContext = new Dictionary<int, ReadOnlyMemory<byte>>(dictionarySize);
                         for (int i = 0; i < dictionarySize; ++i)
                         {
-                            int key = istr.ReadVarInt();
-                            int entrySize = istr.ReadSize();
-                            binaryContext[key] = buffer.AsReadOnlyMemory(istr.Pos, entrySize);
-                            istr.Skip(entrySize);
+                            (int key, ReadOnlyMemory<byte> value) = istr.ReadBinaryContextEntry();
+                            binaryContext[key] = value;
```

Comment:
Should we use Add to detect / reject duplicates?

Reply:
Yes, it would be consistent with how we decode dictionaries.